pybaseball/datasources/fangraphs.py

```diff
@@ -1,44 +1,135 @@
+    COLUMN_NAME_MAPPER: ColumnListMapperFunction = GenericColumnMapper().map_list
 
-    headings_xpath = f"({xpath}/thead//th[contains(@class, 'rgHeader')])[position()>1]/descendant-or-self::*/text()"
-    headings = html_dom.xpath(headings_xpath)
+    def __init__(self):
+        self.html_accessor = HTMLTableProcessor(
+            root_url=self.ROOT_URL,
+            headings_xpath=self.HEADINGS_XPATH,
+            data_cell_xpath=self.DATA_CELLS_XPATH,
+            data_rows_xpath=self.DATA_ROWS_XPATH,
+            table_class=self.TABLE_CLASS,
+        )
 
-    if column_name_mapper:
-        headings = [column_name_mapper(h) for h in headings]
+    def _postprocess(self, data: pd.DataFrame) -> pd.DataFrame:
+        return data
 
-    data_rows_xpath = f"({xpath}/tbody//tr)"
-    data_rows_dom = html_dom.xpath(data_rows_xpath)
-    data_rows = [
-        [
-            postprocessing.try_parse(y, headings[index], known_percentages=known_percentages)
-            for index, y in enumerate(x.xpath('td[position()>1]/descendant-or-self::*/text()'))
-        ]
-        for x in data_rows_dom
-    ]
+    def _validate(self, data: pd.DataFrame) -> pd.DataFrame:
+        return data
 
-    fg_data = pd.DataFrame(data_rows, columns=headings)
+    def fetch(self, start_season: int, end_season: Optional[int] = None, types: FanGraphsStat = None,
+              league: FanGraphsLeague = FanGraphsLeague.ALL, qual: Optional[int] = None, split_seasons: bool = True,
+              month: FanGraphsMonth = FanGraphsMonth.ALL, on_active_roster: bool = False, minimum_age: int = MIN_AGE,
+              maximum_age: int = MAX_AGE, team: str = '0', _filter: str = '', players: str = '',
+              position: FanGraphsPositions = FanGraphsPositions.ALL, max_results: int = 1000000,) -> pd.DataFrame:
 
-    return fg_data
+        types = self.DEFAULT_TYPES if types is None else types
 
+        if start_season is None:
+            raise ValueError(
+                "You need to provide at least one season to collect data for. " +
+                "Try specifying start_season or start_season and end_season."
+            )
 
-def get_fangraphs_tabular_data_from_url(url: str, column_name_mapper: Callable = None, known_percentages: List[str] = []) -> pd.DataFrame:
-    content = requests.get(ROOT_URL + url).content
+        if end_season is None:
+            end_season = start_season
 
-    return get_fangraphs_tabular_data_from_html(
-        content,
-        column_name_mapper=column_name_mapper,
-        known_percentages=known_percentages
-    )
+        assert self.STATS is not None
+
+        url_options = {
+            'pos': position.value,
+            'stats': self.STATS.value,
+            'lg': league.value,
+            'qual': qual if qual is not None else 'y',
+            'type': types,
+            'season': end_season,
+            'month': month.value,
+            'season1': start_season,
+            'ind': '1' if split_seasons else '0',
```

Comment:
aka `str(int(split_seasons))` or for that matter `f'{split_seasons:d}'`, but the whole thing seems unnecessary. Why not just leave this as an int?